pkg/mount/block/blockdev_test.go

```diff
@@ -6,33 +6,259 @@ package block
 
 import (
 	"fmt"
+	"path/filepath"
 	"reflect"
+	"strings"
 	"testing"
 
 	"github.com/u-root/u-root/pkg/pci"
 	"github.com/u-root/u-root/pkg/testutil"
 )
 
-// TestFindMountPointNotExists checks that non existent
-// entry is checked and nil returned
-func TestFindMountPointNotExists(t *testing.T) {
-	LinuxMountsPath = "testdata/mounts"
-	if _, err := GetMountpointByDevice("/dev/mapper/sys-oldxxxxxx"); err == nil {
-		t.Errorf(`GetMountpointByDevice("/dev/mapper/sys-oldxxxxxx") = _, %v, want not nil`, err)
+func TestDebug(t *testing.T) {
+	Debug("foo")
+}
+
+func TestDeviceError(t *testing.T) {
+	// Success paths are covered in vm_test.go
+	_, err := Device("noexist")
+	if err == nil {
+		t.Errorf(`Device("noexist") = _,%v, expect an error`, err)
 	}
 }
 
-// TestFindMountPointValid check for valid output of
-// test mountpoint.
-func TestFindMountPointValid(t *testing.T) {
-	LinuxMountsPath = "testdata/mounts"
-	mountpoint, err := GetMountpointByDevice("/dev/mapper/sys-old")
-	if err != nil {
-		t.Errorf(`GetMountpointByDevice("/dev/mapper/sys-old") = _, %v, want not nil`, err)
+func TestBlockDevString(t *testing.T) {
+	for _, tt := range []struct {
+		name     string
+		blockdev *BlockDev
+		want     []string
+	}{
+		{
+			name: "complete",
+			blockdev: &BlockDev{
+				Name:   "devname",
+				FSType: "sometype",
+				FsUUID: "xxxx",
+			},
+			want: []string{"devname", "sometype", "xxxx"},
+		},
+		{
+			name: "without FSType",
+			blockdev: &BlockDev{
+				Name:   "devname",
+				FsUUID: "xxxx",
+			},
+			want: []string{"devname", "xxxx"},
+		},
+		{
+			name: "without FSType and FsUUID",
+			blockdev: &BlockDev{
+				Name: "devname",
+			},
+			want: []string{"devname"},
+		},
+	} {
+		t.Run(tt.name, func(t *testing.T) {
+			got := tt.blockdev.String()
+			for _, w := range tt.want {
+				if !strings.Contains(got, w) {
+					t.Errorf("String() = %s, does not contain %s", got, w)
+				}
+			}
+
+		})
+	}
+}
+
+func TestBlockDevDevicePath(t *testing.T) {
+	name := "devname"
+	b := &BlockDev{Name: name}
+	want := filepath.Join("/dev", name)
+	got := b.DevicePath()
+	if got != want {
+		t.Errorf("got %s, want %s", got, want)
+	}
+}
+
+func TestBlockDevDeviceName(t *testing.T) {
+	want := "devname"
+	b := &BlockDev{Name: want}
+	got := b.DevName()
+	if got != want {
+		t.Errorf("got %s, want %s", got, want)
+	}
+}
+
+func TestBlockDevGPTTableError(t *testing.T) {
+	// Success paths are covered in vm_test.go
+	tests := []struct {
+		name string
+		dev  *BlockDev
+	}{
+		{
+			name: "Empty BlockDev",
+			dev:  &BlockDev{},
+		},
+		{
+			name: "Not exist",
+			dev:  &BlockDev{Name: "noexist"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			_, err := tt.dev.GPTTable()
+			if err == nil {
+				t.Error("Expect an error")
+			}
```

Comment:
Which errors are expected?

Reply:
I think it is not necessary to specify further here. This error message together with the name of the subtest should give enough context in the test log. 
What would be your suggestion for the message?